### PR TITLE
feat(curriculum): add quiz for JavaScript Events

### LINKED ISSUE #56207: [HACKTOBERFEST 2024] - Events Quiz
> [!IMPORTANT]
> This issue is part of our 2024 Hacktoberfest initiative. If you are interested in working on this issue, refer to our [top-level tasks](https://github.com/freeCodeCamp/freeCodeCamp/issues/56170) to see if this issue is claimed, or to claim it if it is not.

This issue is to create the Events quiz.

The file you should be working on is found in `/curriculum/challenges/english/25-front-end-development/quiz-events`.

**All quiz questions need to relate to one of these topics:**

- Event object
- addEventListener()
- removeEventListener()
- inline event handlers
- change event
- event bubbling/propagation
- event delegation

> [!CAUTION]
> Questions which do not fall under these topics will be rejected.

You are welcome to include code blocks in your questions, but please keep them short (four lines or less).

Please do NOT open a pull request until you have completed all questions. Your quiz should have **exactly 20 questions**.

### PR DESCRIPTION
**Pull request by userAdityaa**
Checklist:
- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #56207

### CODE REVIEW

**Comment by ilenia-magoni** [comment]
You are missing backticks around many code snippets, anything code needs to be wrapped in backticks

**Comment by userAdityaa** [comment]
> You are missing backticks around many code snippets, anything code needs to be wrapped in backticks

Ok, Fixing this.

**Comment by userAdityaa** [comment]
Hey @ilenia-magoni, Can you check the backtick issue now ?

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 1]
```diff
@@ -17,439 +17,440 @@ Answer all of the questions below correctly to pass the quiz.
...
 ---
 
-Placeholder distractor 3
+`onload`
 
 #### --answer--
 
-Placeholder answer
+`onfocus`
+
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:37]:
```suggestion
`onfocus`
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 2]
```diff
@@ -17,439 +17,440 @@ Answer all of the questions below correctly to pass the quiz.
...
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:38]:
```suggestion
`onfocus`

```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 3]
```diff
@@ -17,439 +17,440 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --distractors--
 
-Placeholder distractor 1
+DOMReady
 
 ---
 
-Placeholder distractor 2
+DOMFinished
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:183]:
```suggestion
`DOMFinished`
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 4]
```diff
@@ -17,439 +17,440 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 ---
 
-Placeholder distractor 3
+DocumentReady
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:187]: here

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 5]
```diff
@@ -17,439 +17,440 @@ Answer all of the questions below correctly to pass the quiz.
...
 
 #### --answer--
 
-Placeholder answer
+DOMContentLoaded
```
Comment (ilenia-magoni) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:191]: here

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 6]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+It allows a single listener on a parent instead of multiple on children.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+In general, an event handler is nothing but ___________
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:284]:
```suggestion
In general, an event handler is nothing but a(n) ___________.
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 7]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+The use of single event listener on a parent element to handle events for its children.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the purpose of the stopImmediatePropagation() method in JavaScript event handling?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:350]:
```suggestion
What is the purpose of the `stopImmediatePropagation()` method in JavaScript event handling?
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 8]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+To prevent other listeners on the same element from being called
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the purpose of the event.currentTarget property in JavaScript event handling?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:372]:
```suggestion
What is the purpose of the `event.currentTarget` property in JavaScript event handling?
```

File: curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md [hunk 9]
```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
...
 
-Placeholder answer
+To detect when an element gains focus
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+What is the purpose of the mouseenter event in JavaScript event handling?
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/25-front-end-development/quiz-javascript-events/66edd0ac31fea6e678eb925a.md:438]:
```suggestion
What is the purpose of the `mouseenter` event in JavaScript event handling?
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Nice work! We've got a few things to resolve before this is ready.

**Comment by userAdityaa** [comment]
Hey @naomi-lgbt, Done with the changes.

**Review by naomi-lgbt — APPROVED** [review]
Nice work! Think we're all good!

**Comment by userAdityaa** [comment]
Hey, Why is this one test failing can anyone help me ?

**Comment by naomi-lgbt** [comment]
It's a Euler challenge timing out - not related to your PR. 💜

**Comment by userAdityaa** [comment]
Any update on this @naomi-lgbt, @gikf, @ilenia-magoni

**Comment by a2937** [comment]
> Do NOT copy questions directly from other sources, even if paraphrased. Questions you submit should be your own creations. This includes using generative AI to create questions - please do NOT use AI (ChatGPT, Claude, etc) to create questions.

**Comment by naomi-lgbt** [comment]
That's disappointing.
I'll release the issue back to other contributors.

**Pull request closed without merging**